packages/vite/src/index.ts

```diff
@@ -1,5 +1,5 @@
 import { basename, extname } from 'node:path'
-import { join } from 'node:path/posix'
+import urlJoin from 'url-join'
```

Comment:
I don't think we need to add a dependency for such a simple piece of functionality. Can you just add a method here to do it instead?

Reply:
I changed it to string concatenation. This should work fine, because Vite makes sure that the origin option never ends with a '/'.